syntax = "proto3";


service ShotDetector {
    rpc copy_video(stream VideoRequest) returns (VideoResponse);
    rpc get_shots(ShotRequest) returns (ShotResponse);
}

message VideoRequest {
    string video_id = 1;
    bytes video_encoded = 2;
}

message VideoResponse {
    bool success = 1;
}

message ShotRequest {
    string video_id = 1;
}

message ShotResponse {
    bool success = 1;
    repeated Shot shots = 2;
}

message Shot {
    int32 shot_id = 1;
    int32 start_frame = 2;
    int32 end_frame = 3;
}
// EOF